app/core/Engine/Engine.ts

```diff
@@ -1751,6 +1764,22 @@ export class Engine {
         }
       },
     );
+
+    this.appStateListener = AppState.addEventListener(
+      'change',
+      (state: AppStateStatus) => {
+        if (state === 'active') {
+          this.controllerMessenger.call('SnapController:setClientActive', true);
+        }
+
+        if (state === 'background') {
+          this.controllerMessenger.call(
+            'SnapController:setClientActive',
+            false,
+          );
+        }
```

Comment:
Could we simplify this to something like this?
```suggestion
        this.controllerMessenger.call('SnapController:setClientActive', state === 'active');
```

Reply:
There is a couple of other state which I'm not sure we want to react to 🤔 I'll try and see.

Reply:
Have you tested that this actually triggers in time for the Snap to receive the message?

Reply:
Tried with your suggested implementation but we need to specify the state to trigger `onInactive` otherwise we trigger it multiple time

<img width="889" height="302" alt="Screenshot 2025-08-06 at 11 51 06" src="https://github.com/user-attachments/assets/38bb8d84-4c05-4d92-9606-35f9f329eb02" />